app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,18 @@
+module Foreman
+  class Deprecation
+    #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
+    def self.deprecation_warning(foreman_version_deadline, info, callstack = nil)
+      callstack ||= caller
+      if (foreman_version_deadline == "API")
```

Comment:
Not sure there's any point in this, just keep the deprecation warning in the api_deprecation_warning method.